### DISCUSSION TITLE
Calibrating Namron Thermostat 16A Plug and weird behavior regarding calibration

### DISCUSSION THREAD

**Discussion by jkaberg** [zwave-js/zwave-js#7215] [Request Support / Investigate Issue]
### Checklist

- [X] I have read and followed the above instructions
- [X] I have checked the [troubleshooting section](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index) and my problem is **not** described there.

- [X] I have read the [changelog](https://github.com/zwave-js/node-zwave-js/blob/master/CHANGELOG.md) and my problem was **not** mentioned there or the fix **did not work**.

### Describe the issue

What is happening?
I'm trying to calibrate my thermostat plug but I'm seeing wierd behaviour when doing so, see [video](https://drive.google.com/file/d/1wJcQ0wNBJiadNHgzrsJajvDYTga2pTyU/view?usp=sharing). Essentially it doesnt increment the temperature with the calibrated value, but just some random? number.

I have 7 of them and the same ting is happening to all of them.

What did you expect to happen instead?
When calibrating it would actually adjust the reported temperature to temp+calibrated value.

 See video above 😄 

Anything else we should know?
Heres device info from the manual
![image](https://github.com/user-attachments/assets/41cc148c-4761-4685-b278-2895898a7df1)

[Z-wave JS Device DB link](https://devices.zwave-js.io/?jumpTo=0x0438:0x0200:0xd004:0.0)

### Software versions

zwave-js-ui: 9.21.0.24dc718
zwave-js: 13.4.0

### Z-Wave Controller (Stick/Dongle/...)

Aeon Labs ZW090 Z‐Stick Gen5 USB Controller

### Device information

Manufacturer: Namron
Model name: Thermostat plugg 16A
Node ID: 66

### Checklist

- [X] I [made sure](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index?id=rule-1-use-the-driver-log-luke) to provide a **driver log** on level **debug**.

- [ ] The log includes a re-interview of the problematic device _(if applicable)_.

- [X] The log includes the problematic interaction with the device _(if applicable)_.

- [X] I provided the node ID of the problematic device _(if applicable)_.

### Upload Logfile

[zwavejs_2024-10-02.log](https://github.com/user-attachments/files/17228354/zwavejs_2024-10-02.log)

**Comment by jkaberg** [comment]
[This issue](https://github.com/zwave-js/node-zwave-js/issues/6792) is somewhat? related, however I'm not trying to set the target temperature but the calibration of the temperature reading. Note that the device here is not the same one as I'm referring to (which is [this one](https://github.com/zwave-js/node-zwave-js/blob/master/packages/config/config/devices/0x0438/4512725.json))

**Comment by jkaberg** [comment]
@AlCalzone do you have any ideas? I'm thinking faulty firmware but before I return these I thought I'd see if its a config problem first, however evaluating that is out of my depth

**Reply by AlCalzone** [reply]
The Z-Wave communication looks normal, so if the temperature around the device was not fluctuating wildly while you did your tests, I'm thinking there's something wrong with the device.

**Comment by jkaberg** [comment]
Thing is I got 7 of them and all of them behave the same in terms of calibration which made me think perhaps there is an (zwave-js) config error as there are no report elsewhere on this wonky behavior 

As you can see from the video when I set the calibration setting it just throws off really weird temperature numbers, eg:
| Calibration | (new) Temperature | Expected Temperature |
| ----------- | ----------------- | -------------------- |
| 0           | 18.42             | 18.42                |
| 10          | 19.46             | 19.42                |
| 20          | 17.9              | 20.42                |
| 30          | 18.9              | 21.42                |
| 40          | 17.37             | 22.42                |
| 50          | 18.35             | 23.42                |

**Reply by AlCalzone** [reply]
Yeah it's weird. The communication here is done using Supervision, so the device tells us that it accepted and executed the config change. This rules out incorrect value sizes in our config files etc.

**Reply by jkaberg** [reply]
@AlCalzone sorry to bother you with this again but after I did an RMA the manufacturer tested the same devices with Homey and they're telling me everything (including temperature calibration) is working as expected. 

Is there anything I can do or ask for that would help us solve this?

**Reply by AlCalzone** [reply]
It might help if they could share Zniffer traces or some other human-readable way that includes the communication between the controller and the thermostat for:
- changing the calibration parameters
- the devices' temperature reports afterwards

I guess it makes sense to do this once with Supervision for the configuration change, and once without, just to exclude bugs in their Supervision implementation.
This can be done with the **PC Controller** tool from Silicon Labs.

**Comment by jkaberg** [comment]
@bruberg just wondering seeing as you did the initial implementation - are you seeing this calibration behavior with your devices?

**Reply by bruberg** [reply]
Sorry, never tried calibrating so I haven't seen (nor looked for) this issue.

**Reply by jkaberg** [reply]
Would you mind testing it? Just to verify if its my devices
